tools/walletextension/wallet_extension.go

```diff
@@ -153,105 +131,111 @@ func (we *WalletExtension) handleHTTPEthJSON(resp http.ResponseWriter, req *http
 		return
 	}
 
-	// We unmarshall the JSON request.
-	var reqJSONMap map[string]interface{}
+	// We unmarshall the JSON request
```

Comment:
@BedrockSquirrel I'd be tempted to put all the unmarshalling into a method, that returns either the ID, the method and the params, or a struct containing all three. That will help prevent the existing method from getting too long.